clinics/merge/clinicians.go

```diff
@@ -138,6 +144,10 @@ func (s *SourceClinicianMergePlanner) Plan(ctx context.Context) (ClinicianPlan,
 		}
 	}
 
+	if plan.ClinicianAction == ClinicianActionMove && len(s.target.MembershipRestrictions) > 0 {
```

Comment:
If the clinician has the appropriate email domain, is that then something we should allow?